### PR TITLE
13993 - disable Refresh in Player if gpid errors, delete orphan pieces when r…

### PR DESCRIPTION
**Pull request by claudiociardelli**
…efreshing

### CODE REVIEW

**Comment by claudiociardelli** [comment]
Bug 13993 is fixed by PR680. If someone wants to test, use the latest Pursuit of Glory module in the module pages

File: vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java [hunk 1]
```diff
@@ -228,11 +232,14 @@ public void execute(Set<String> options, Command command) throws IllegalBuildExc
...
-        // Any errors should have been resolved by the GpId check at startup, so
-        // this error indicates
-        // a bug in GpIdChecker.fixErrors().
+        // Any gpid errors should have been resolved by the GpId check when the editor is run.
+        // If a module created before gpIDChecker was setup is run on a vassal version with gmIDChecker
+        // is run in the player, errors might still be present.
+        // Inform user that he must upgrade the module to the latest vassal version before running Refresh
         gpIdChecker = null;
-        throw new IllegalBuildException("GameRefresher.execute: gpIdChecker has errors"); //$NON-NLS-1$
+        log("Unable to run Refresh, module was saved with older vassal version. Edit and save module with latest vassal version first. "); //$NON-NLS-1$
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java:240]: Needs i18n

File: vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java [hunk 2]
```diff
@@ -262,6 +269,13 @@ private void processGamePiece(GamePiece piece, Command command) {
     if (map == null) {
 //      logger.error("Can't refresh piece " + piece.getName() + + "(" + piece.getId()+ "): No Map"); //$NON-NLS-1$
       log("Can't refresh piece " + piece.getName() + "(" + piece.getId() + "): No Map"); //$NON-NLS-1$
+      // If Option "Delete pieces with no map" is set to true. Get rid of this piece
+      if (isDeleteNoMap()) {
+        log("Deleting " + piece.getName() + "(" + piece.getId() + ") from game"); //$NON-NLS-1$
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java:274]: Needs i18n of strings

**Review by BrentEaston — COMMENTED** [review]
Hi Claudio,

Overall, looks good.

Don't worry about checking the gpid's up front and disabling the run button as I suggested in Discord, this works just fine. Just make sure all messages presented to the users via an i18n key. Looks like there are several instances that where already untranslated before you started. These where once Vassal logfile only message, but are now being presented to the user so now need a translation.

**Review by BrentEaston — CHANGES_REQUESTED** [review]
...

**Review by BrentEaston — APPROVED** [review]

**Pull request merged**